### PR TITLE
Move npm modules from emitter folder to outside folder

### PR DESCRIPTION
**Pull request by pshao25**
1. Currently, we have two node_modules. One is in the root folder and the other is in the emitter folder, which is confusing. This change merges them together into the root folder.

2. Currently we need to run `.\eng\Generate.ps1 cadl-project -reset` to apply the latest change in the emitter, which costs. This change links the `@azure-tools/cadl-csharp` to the source code.

### CODE REVIEW

File: src/CADL.Extension/Emitter.Csharp/package.json [hunk 1]
```diff
@@ -36,14 +36,14 @@
         "dist/**"
     ],
     "devDependencies": {
+        "@azure-tools/cadl-azure-core": "0.6.0",
```
Comment (chunyu3) [inline src/CADL.Extension/Emitter.Csharp/package.json:39]: It is better to keep the order as before, just to update the version. In that way, the diff may more clear about what is changed.

File: eng/Generation.psm1 [hunk 2]
```diff
@@ -93,42 +93,17 @@ function Invoke-CadlSetup()
...
-            $package = $packages;
-            if ($packages.count -gt 1)
-            {
-                $package = $packages[0]
-            }
-            npm install $package
-        }
-        git checkout $repoRoot/package.json
-        git checkout $repoRoot/package-lock.json
+        npm run build
```
Comment (chunyu3) [inline eng/Generation.psm1:99]: Will it use the emitter version build locally?

**Review by chunyu3 — COMMENTED** [review]
Why delete package-lock.json under src/CADL.Extension/Emitter.Csharp?

File: eng/Generation.psm1 (hunk 2 above)
Comment (pshao25) [reply eng/Generation.psm1:99]: You mean in CI? It will use the build in the build agent.

**Comment by pshao25** [comment]
> Why delete package-lock.json under src/CADL.Extension/Emitter.Csharp?

There is no place we need that. Now we just need to run `npm install` from the root folder and all the dependencies are installed in the root folder.

File: src/CADL.Extension/Emitter.Csharp/package.json (hunk 1 above)
Comment (pshao25) [reply src/CADL.Extension/Emitter.Csharp/package.json:39]: I'm trying to make it in alphabetical order.

**Review by chunyu3 — APPROVED** [review]

**Pull request merged**